crates/tests/src/e2e/ibc_tests.rs

```diff
@@ -526,6 +528,115 @@ fn pgf_over_ibc() -> Result<()> {
     Ok(())
 }
 
+// Test fee payment with an ibc token
+//
+// 1. Submit governance proposal to allow fee payment with the IBC token
+// 2. Transfer some IBC tokens from gaia
+// 3. Transparent transfer in Namada with ibc token gas payment
+#[test]
+fn fee_payment_with_ibc_token() -> Result<()> {
+    const PIPELINE_LEN: u64 = 2;
+    let update_genesis =
+        |mut genesis: templates::All<templates::Unvalidated>, base_dir: &_| {
+            genesis.parameters.parameters.epochs_per_year =
+                epochs_per_year_from_min_duration(30);
+            genesis.parameters.ibc_params.default_mint_limit =
+                Amount::max_signed();
+            genesis.parameters.gov_params.min_proposal_grace_epochs = 1;
+            genesis
+                .parameters
+                .ibc_params
+                .default_per_epoch_throughput_limit = Amount::max_signed();
+            // Artificially increase the gas scale to allow for fee payment with
+            // the limited ibc tokens available
+            genesis.parameters.parameters.gas_scale = 10_000_000;
+            setup::set_validators(1, genesis, base_dir, |_| 0, vec![])
+        };
+    let (ledger, gaia, test, test_gaia) = run_namada_gaia(update_genesis)?;
+    let _bg_ledger = ledger.background();
+    let _bg_gaia = gaia.background();
+
+    // Proposal on Namada
+    // Delegate some token
+    delegate_token(&test)?;
+    let rpc = get_actor_rpc(&test, Who::Validator(0));
+    let mut epoch = get_epoch(&test, &rpc).unwrap();
+    let delegated = epoch + PIPELINE_LEN;
+    while epoch < delegated {
+        sleep(10);
+        #[allow(clippy::disallowed_methods)]
+        let new_epoch = get_epoch(&test, &rpc).unwrap_or_default();
+        epoch = new_epoch;
```

Comment:
nit: use `epoch_sleep` helper here and below

Reply:
Done in d9e590be6d2eff6e4c7d765a776b69103a7cced3, I've also changed the other tests in this module to use this function

Reply:
nice thx!